### ISSUE TITLE
[GKE V2] Cluster stuck in updating state "Waiting for nodes"

### ISSUE THREAD

**Issue by aaronRancher** [rancher/rancher#31986]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
On Rancher `master-head` `version f5393c2`
Create a google cloud credential.
From GKE console, create a default k8s cluster.
From Rancher, import the cluster from the API UI.
Input cluster name and config in `gkeConfig`:
```
{
  "clusterName": "aaron-test",
  "displayName": "aaron-test",
  "googleCredentialSecret": "cattle-global-data:cc-dg229",
  "imported": true,
  "projectID": "rancher-qa",
  "region": "us-central1-c"
}
```
Import cluster

**Result:**
The cluster is provisioned successfully, but becomes stuck in updating state: `Waiting for nodes`

**Other details that may be helpful:**
This is also seen in hosted EKS V2 clusters created from the API.
EKS operator version: `v1.0.1-rc6`

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): `master-head` `version f5393c2`
- Installation option (single install/HA): Single install

**Comment by aaronRancher** [comment]
Tested on Rancher `v2.5-head` `version 7426ba8`
EKS operator version: `v1.0.1-rc6`

Tested import use case:
Imported cluster is stuck in updating state with: `Waiting for nodes`

Tested hosted cluster with the following gkeConfig:
[gke5.json.zip](https://github.com/rancher/rancher/files/6280638/gke5.json.zip)
The cluster is stuck in updating state with: `Waiting for nodes`
The cluster is seen in GKE console with an unschedulable pod:
```
Problem
Your cluster has 1 unschedulable pod

Details
Unschedulable pods
calico-typha-7c595b658-x2v26
```

EKS operator logs:
c-85qqc - imported cluster
c-8m9j8 - hosted cluster
```
time="2021-04-08T17:10:03Z" level=info msg="Starting /v1, Kind=Secret controller"
time="2021-04-08T17:10:04Z" level=error msg="Error recording gkecc [] failure message: resource name may not be empty"
time="2021-04-08T17:10:18Z" level=info msg="cluster [c-85qqc] is imported, will not delete GKE cluster"
time="2021-04-08T17:21:17Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:21:17Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:21:17Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:21:47Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:22:17Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:22:47Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:23:17Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:23:47Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:24:17Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:24:47Z" level=info msg="waiting for cluster [c-8m9j8] to finish creating"
time="2021-04-08T17:25:17Z" level=info msg="Cluster aaron-gke is running"
```

**Comment by aaronRancher** [comment]
This issue is validated on Rancher `v2.5-head` `version d1083bc`
GKE operator version: `gke-operator:v1.0.1-rc6`

Hosted cluster is provisioned successfully through the API. 
Imported cluster is provisioned successfully through the API.

**Comment by aaronRancher** [comment]
This behavior is seen on Rancher `master-head` `version b202f20`
GKE operator version: `gke-operator:v1.0.1-rc6`

Imported and hosted clusters are stuck updating with: `Waiting for nodes`

 Reopening for a fix on `master-head`

**Comment by cmurphy** [comment]
@aaronRancher can you confirm the version of `rancher-operator` that's deployed? The fix is in v0.1.4-rc3:

```
kubectl -n rancher-operator-system get deploy rancher-operator -o jsonpath='{.spec.template.spec.containers[0].image}'
rancher/rancher-operator:v0.1.4-rc3
```

**Comment by aaronRancher** [comment]
On `v2.5-head` `version d1083bc`:
Rancher Operator version: `rancher-operator:v0.1.4-rc3`

On `master-head` `version b202f20`:
Rancher Operator version: `rancher-operator:v0.2.0-alpha02`

**Comment by cmurphy** [comment]
@aaronRancher this should be fixed in master now with rancher-operator v0.2.0-alpha03

**Comment by aaronRancher** [comment]
This issue is validated on Rancher `master-head` `version 39d572a`
GKE operator version: `gke-operator:v1.0.1-rc6`
Rancher Operator version: `rancher-operator:v0.2.0-alpha03`

Hosted cluster is provisioned successfully through the API.
Imported cluster is provisioned successfully through the API.
